### DISCUSSION TITLE
siliconcompiler._common.SiliconCompilerError: Invalid keypath ('option', 'scpath'): unexpected key: scpath

### DISCUSSION THREAD

**Discussion by MuraliGaneshCh** [The-OpenROAD-Project/OpenROAD-flow-scripts#2077] [General]
trying to run nangate45 design using makefile.py in openroad flow. when running the following commands:

export SCPATH=./scripts/sc
python3 Makefile.py -DESIGN_CONFIG=./designs/nangate45/gcd/config.mk

i am getting these errors:

Traceback (most recent call last):
  File "/home/mchittap/ORFS/OpenROAD-flow-scripts/flow/Makefile.py", line 138, in <module>
    main()
  File "/home/mchittap/ORFS/OpenROAD-flow-scripts/flow/Makefile.py", line 68, in main
    chip.set('option', 'scpath' , scdir)
  File "/home/mchittap/.local/lib/python3.10/site-packages/siliconcompiler/core.py", line 887, in set
    self.error(e)
  File "/home/mchittap/.local/lib/python3.10/site-packages/siliconcompiler/core.py", line 3522, in error
    raise SiliconCompilerError(msg) from None
siliconcompiler._common.SiliconCompilerError: Invalid keypath ('option', 'scpath'): unexpected key: scpath

can someone suggest how to solve these errors. i have not changed any files inside the makefile.py and updated to latest siliconcompiler.

**Comment by maliberty** [comment]
@gadfort any suggestions?

**Comment by mi-and-core** [comment]
@MuraliGaneshCh I'm not involved, but to my knowledge $SCPATH (and ['option', 'scpath'] ) are no longer supported since Siliconcompiler 0.17.0 (November? 2023). Looks like Siliconcompiler is a "moving target". For an alternative, see https://docs.siliconcompiler.com/en/stable/reference_manual/schema.html#param-package-source-ref:   chip.set('package', 'source' ... That should work for the current siliconcompiler 0.23.1

**Comment by gadfort** [comment]
@MuraliGaneshCh I don't know if this code has been maintained since it was merged in 2years ago it seems. It's unclear to me that this would work with a newer version of SC, you can always downgrade to a compatible version of SC.
